### PR TITLE
Fix types in doc comments part 1

### PR DESCRIPTION
**Pull request by tmotyl**
This commit fixes many issues with doc comments in Magento,
to ease the pain of reading the code and debugging.

This patch does not fix everything, but quite a lot.
Once this is accepted I can provide more.

What is changed:
- add missing doc comments for some params
- change "decimal" to "float"
- fix some non existing class names (like "Mage_Sales_Model_Invoice")
- fix wrong variable names in doc comments
- fix wrong class names (e.g. Mage_Core_Model_Cache instead of Mage_Core_Model_Cache
in Mage_Core_Model_App)
- remove `@return` doc comment for methods which are not returning

Let me know whether this kind of cleanup is wanted. If yes, i will most probably split this PR into few chunks (e.g. move decimal->float to a separate PR) to make it easier for a review.

I'm also open for doing a one-to-one review session, I think 30min is enough to go through all the changed places and verify them.

### CODE REVIEW

**Comment by colinmollenhour** [comment]
Looking great! Definitely lots of vardoc errors in Magento core so thanks for working on this!

A few comments/suggestions, not that they need to be implemented for my approval, but just to get them out there for consideration before too much time is spent:

### Method chaining: `@return $this`

PhpStorm supports `@return $this` which is better for method chaining than e.g. `@return Varien_Object` because the latter breaks method chaining for methods that are not part of the abstract class. E.g.:

```php
Mage::getModel('admin/user')->setData(...)->save();
```

In the case above `setData` has `@return Varien_Object` and so `save()` is not recognized because `Varien_Object` does not have a `save()` method. Using `@return $this` solves this.

I'm pretty sure an awk script could automatically fix all of these issues fairly easily.

### Models: `@method ... getResource()`

I like to add this for models so that resource methods can be resolved. For example on `Mage_Admin_Model_User` adding `@method Mage_Admin_Model_Resource_User getResource()` fixes this code:

```php
$user = Mage::getModel('admin/user')->getResource()->loadByUsername($username);
```

I would actually write this differently myself, but nevertheless it is a common pattern in Magento code and is just to demonstrate the point.

```php
$user = Mage::getResourceSingleton('admin/user')->loadByUsername($username);
```

### Collections: `@method ...`

For collections I like to add the following, again using `admin/user` as an example:

```
 * @method Mage_Admin_Model_User[] getIterator()
 * @method Mage_Admin_Model_User getFirstItem()
 * @method Mage_Admin_Model_User[] getItems()
 * @method Mage_Admin_Model_User getItemById($value)
 * @method Mage_Admin_Model_User[] getItemsByColumnValue($column, $value)
 * @method Mage_Admin_Model_User getItemByColumnValue($column, $value)
```

The most useful of these probably being `getIterator()` since this allows `foreach` loops to automatically resolve the iterator object type with no hinting:

```php
$adminUserCollection = Mage::getResourceModel('admin/user_collection');
foreach ($adminUserCollection as $adminUser) {
    echo $adminUser->getEmail();
}
```

**Comment by colinmollenhour** [comment]
I forgot that Magento already has the `getResource` annotations in the models, so nothing to do there actually..

Here is a script which adds the `@method ` annotations to the collection classes. I haven't run it yet on a clean environments since I don't have one handy at the moment but I believe it will do 99% of the work.
https://gist.github.com/colinmollenhour/75f65a63e5f66adc8639aa8b9b1eef5f

I submitted a PR for the `@return $this` update at #624

**Comment by Flyingmana** [comment]
I like the changes. So the plan is to split this PR up into separate ones?
You forgott to mention there is one change which fixes a incomplete closing tag

File: app/code/core/Mage/CatalogIndex/Model/Resource/Data/Abstract.php [hunk 1]
```diff
@@ -128,7 +128,7 @@ public function getAttributeData($products, $attributes, $store)
      * @param string $table
      * @param string $idField
      * @param string $whereField
-     * @param int $id
+     * @param int|array $id
      * @param array $additionalWheres
      * @return mixed
```
Comment (LeeSaferite) [inline app/code/core/Mage/CatalogIndex/Model/Resource/Data/Abstract.php]:
```suggestion
     * @return array
```

File: app/code/core/Mage/Core/Block/Abstract.php [hunk 2]
```diff
@@ -1079,7 +1079,7 @@ public function getHelper($type)
      * Returns helper object
      *
      * @param string $name
-     * @return Mage_Core_Block_Abstract
+     * @return Mage_Core_Helper_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Core/Block/Abstract.php]: Again, not strictly true

File: app/code/core/Mage/Core/Controller/Front/Action.php [hunk 3]
```diff
@@ -105,7 +105,7 @@ public function __()
      *                              that case
      * @param string $contentType
      * @param int $contentLength    explicit content length, if strlen($content) isn't applicable
-     * @return Mage_Adminhtml_Controller_Action
+     * @return Mage_Core_Controller_Front_Action
```
Comment (LeeSaferite) [inline app/code/core/Mage/Core/Controller/Front/Action.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Core/Model/App.php [hunk 4]
```diff
@@ -469,7 +469,7 @@ protected function _initRequest()
      *
      * @param string $scopeCode code of default scope (website/store_group/store code)
      * @param string $scopeType type of default scope (website/group/store)
-     * @return unknown_type
+     * @return Mage_Core_Model_App
```
Comment (LeeSaferite) [inline app/code/core/Mage/Core/Model/App.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Core/Model/App.php [hunk 5]
```diff
@@ -517,7 +517,9 @@ public function getCookie()
     /**
      * Check get store
      *
+     * @param string $type
      * @return Mage_Core_Model_App
```
Comment (LeeSaferite) [inline app/code/core/Mage/Core/Model/App.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Core/Model/Store.php [hunk 6]
```diff
@@ -393,7 +393,7 @@ public function getConfig($path)
      *
      * Method provide cache configuration data without loading store config XML
      *
-     * @return Mage_Core_Model_Config
+     * @return Mage_Core_Model_Store
```
Comment (LeeSaferite) [inline app/code/core/Mage/Core/Model/Store.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Customer/Model/Address/Abstract.php [hunk 7]
```diff
@@ -153,8 +153,8 @@ public function setStreetFull($street)
     /**
      * set address street informa
      *
-     * @param unknown_type $street
-     * @return unknown
+     * @param array|string $street
+     * @return Mage_Customer_Model_Address_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Customer/Model/Address/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Customer/Model/Resource/Address.php [hunk 8]
```diff
@@ -90,7 +90,7 @@ public function getCustomerId($object)
      *
      * @param Mage_Customer_Model_Address $object
      * @param int $id
-     * @return Mage_Customer_Model_Address
+     * @return Mage_Customer_Model_Resource_Address
```
Comment (LeeSaferite) [inline app/code/core/Mage/Customer/Model/Resource/Address.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Eav/Model/Entity/Abstract.php [hunk 9]
```diff
@@ -504,7 +504,7 @@ public function isPartialSave($flag = null)
     /**
      * Retrieve configuration for all attributes
      *
-     * @return Mage_Eav_Model_Entity_Attribute_Abstract
+     * @return Mage_Eav_Model_Entity_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Eav/Model/Entity/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Eav/Model/Entity/Abstract.php [hunk 10]
```diff
@@ -1501,7 +1501,7 @@ protected function _prepareValueForSave($value, Mage_Eav_Model_Entity_Attribute_
      * @param   Varien_Object $object
      * @param   string $table
      * @param   array $info
-     * @return  Varien_Object
+     * @return  Mage_Eav_Model_Entity_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Eav/Model/Entity/Abstract.php]:
```suggestion
     * @return  $this
```

File: app/code/core/Mage/Eav/Model/Entity/Attribute/Source/Abstract.php [hunk 11]
```diff
@@ -53,7 +53,7 @@ abstract class Mage_Eav_Model_Entity_Attribute_Source_Abstract
      * Set attribute instance
      *
      * @param Mage_Eav_Model_Entity_Attribute_Abstract $attribute
-     * @return Mage_Eav_Model_Entity_Attribute_Frontend_Abstract
+     * @return Mage_Eav_Model_Entity_Attribute_Source_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Eav/Model/Entity/Attribute/Source/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php [hunk 12]
```diff
@@ -177,7 +177,9 @@ protected function _initSelect()
      * Standard resource collection initalization
      *
      * @param string $model
-     * @return Mage_Core_Model_Mysql4_Collection_Abstract
+     * @param null|string $entityModel
+     * @return Mage_Eav_Model_Entity_Collection_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 13]
```diff
@@ -385,7 +385,7 @@ public function getInfoInstance()
     /**
      * Validate payment method information object
      *
-     * @return Mage_Payment_Model_Abstract
+     * @return Mage_Payment_Model_Method_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 14]
```diff
@@ -410,7 +410,7 @@ public function validate()
      * @param Varien_Object $payment
      * @param float $amount
      *
-     * @return Mage_Payment_Model_Abstract
+     * @return Mage_Payment_Model_Method_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 15]
```diff
@@ -426,7 +426,7 @@ public function order(Varien_Object $payment, $amount)
...
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 16]
```diff
@@ -442,7 +442,7 @@ public function authorize(Varien_Object $payment, $amount)
...
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 17]
```diff
@@ -486,7 +486,7 @@ public function processBeforeRefund($invoice, $payment)
...
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 18]
```diff
@@ -516,7 +516,7 @@ public function processCreditmemo($creditmemo, $payment)
      *
      * @param Varien_Object $payment
      *
-     * @return Mage_Payment_Model_Abstract
+     * @return Mage_Payment_Model_Method_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 19]
```diff
@@ -542,7 +542,7 @@ public function processBeforeVoid($invoice, $payment)
...
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 20]
```diff
@@ -625,7 +625,7 @@ public function getConfigData($field, $storeId = null)
      * Assign data to info model instance
      *
      * @param   mixed $data
-     * @return  Mage_Payment_Model_Info
+     * @return  Mage_Payment_Model_Method_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 21]
```diff
@@ -641,7 +641,7 @@ public function assignData($data)
    /**
      * Parepare info instance for save
      *
-     * @return Mage_Payment_Model_Abstract
+     * @return Mage_Payment_Model_Method_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Payment/Model/Method/Abstract.php [hunk 22]
```diff
@@ -741,7 +741,7 @@ public function isApplicableToQuote($quote, $checksBitMask)
      * @param string $paymentAction
      * @param object $stateObject
      *
-     * @return Mage_Payment_Model_Abstract
+     * @return Mage_Payment_Model_Method_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Payment/Model/Method/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Sales/Block/Items/Abstract.php [hunk 23]
```diff
@@ -58,7 +58,7 @@ protected function _construct()
      * @param   string $type
      * @param   string $block
      * @param   string $template
-     * @return  Mage_Checkout_Block_Cart_Abstract
+     * @return  Mage_Sales_Block_Items_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Sales/Block/Items/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 24]
```diff
@@ -152,7 +152,7 @@ protected function _roundTotals(Mage_Sales_Model_Quote_Address $address)
      * Collect tax totals for quote address
      *
      * @param   Mage_Sales_Model_Quote_Address $address
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 25]
```diff
@@ -378,7 +378,7 @@ protected function _calculateShippingTaxByRate(
      *
      * @param   Mage_Sales_Model_Quote_Address $address
      * @param   Varien_Object $taxRateRequest
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 26]
```diff
@@ -407,7 +407,8 @@ protected function _calculateShippingTax(Mage_Sales_Model_Quote_Address $address
      * Calculate address tax amount based on one unit price and tax amount
      *
      * @param   Mage_Sales_Model_Quote_Address $address
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @param $taxRateRequest
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 27]
```diff
@@ -509,7 +510,7 @@ protected function _unitBaseProcessItemTax(
      * @param   array $taxGroups
      * @param   string $taxId
      * @param   boolean $recalculateRowTotalInclTax
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 28]
```diff
@@ -1470,7 +1473,7 @@ protected function _saveAppliedTaxes(Mage_Sales_Model_Quote_Address $address,
      * Add tax totals information to address object
      *
      * @param   Mage_Sales_Model_Quote_Address $address
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 29]
```diff
@@ -1399,7 +1402,7 @@ protected function _deltaRound($price, $rate, $direction, $type = 'regular')
      * Recalculate parent item amounts base on children data
      *
      * @param   Mage_Sales_Model_Quote_Item_Abstract $item
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 30]
```diff
@@ -1023,7 +1024,9 @@ protected function _totalBaseProcessItemTax(
      * @param   Mage_Sales_Model_Quote_Item_Abstract $item
      * @param   float $rate
      * @param   array $taxGroups
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @param  $taxId
+     * @param bool $recalculateRowTotalInclTax
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 31]
```diff
@@ -908,7 +909,7 @@ protected function _calcRowTaxAmount(
      *
      * @param   Mage_Sales_Model_Quote_Address $address
      * @param   Varien_Object $taxRateRequest
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 32]
```diff
@@ -755,7 +756,7 @@ protected function _rowBaseProcessItemTax($address, $item, $taxRateRequest, &$it
      * @param   array $taxGroups
      * @param   string $taxId
      * @param   boolean $recalculateRowTotalInclTax
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php [hunk 33]
```diff
@@ -657,7 +658,7 @@ protected function _calcUnitTaxAmount(
      *
      * @param   Mage_Sales_Model_Quote_Address $address
      * @param   Varien_Object $taxRateRequest
-     * @return  Mage_Tax_Model_Sales_Total_Quote
+     * @return  Mage_Tax_Model_Sales_Total_Quote_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Tax/Model/Sales/Total/Quote/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Usa/Model/Shipping/Carrier/Abstract/Backend/Abstract.php [hunk 34]
```diff
@@ -76,7 +76,7 @@ public function __construct()
      * Check for presence in array with allow value.
      *
      * @throws Mage_Core_Exception
-     * @return Mage_Usa_Model_Shipping_Carrier_Ups_Backend_FreeShipment
+     * @return Mage_Usa_Model_Shipping_Carrier_Abstract_Backend_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/Usa/Model/Shipping/Carrier/Abstract/Backend/Abstract.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Weee/Model/Attribute/Backend/Weee/Tax.php [hunk 35]
```diff
@@ -44,7 +44,7 @@ protected function _getResource()
      * Validate data
      *
      * @param   Mage_Catalog_Model_Product $object
-     * @return  this
+     * @return  Mage_Weee_Model_Attribute_Backend_Weee_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Weee/Model/Attribute/Backend/Weee/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Weee/Model/Attribute/Backend/Weee/Tax.php [hunk 36]
```diff
@@ -76,7 +76,7 @@ public function validate($object)
      * Assign WEEE taxes to product data
      *
      * @param   Mage_Catalog_Model_Product $object
-     * @return  Mage_Catalog_Model_Product_Attribute_Backend_Weee
+     * @return  Mage_Weee_Model_Attribute_Backend_Weee_Tax
```
Comment (LeeSaferite) [inline app/code/core/Mage/Weee/Model/Attribute/Backend/Weee/Tax.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/Widget/Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php [hunk 37]
```diff
@@ -63,7 +63,7 @@ public function render(Varien_Data_Form_Element_Abstract $element)
      * Setter
      *
      * @param Varien_Data_Form_Element_Abstract $element
-     * @return
+     * @return Mage_Widget_Block_Adminhtml_Widget_Instance_Edit_Tab_Main_Layout
```
Comment (LeeSaferite) [inline app/code/core/Mage/Widget/Block/Adminhtml/Widget/Instance/Edit/Tab/Main/Layout.php]:
```suggestion
     * @return $this
```

File: app/code/core/Mage/XmlConnect/Model/Preview/Abstract.php [hunk 38]
```diff
@@ -101,7 +101,7 @@ public function getApplicationModel()
      * Set active tab
      *
      * @param string $tab
-     * @return Mage_XmlConnect_Block_Adminhtml_Mobile_Preview_Tabitems
+     * @return Mage_XmlConnect_Model_Preview_Abstract
```
Comment (LeeSaferite) [inline app/code/core/Mage/XmlConnect/Model/Preview/Abstract.php]:
```suggestion
     * @return $this
```

File: app/design/adminhtml/default/default/template/catalog/product/edit/options/option.phtml [hunk 39]
```diff
@@ -193,7 +193,7 @@ var productOptionType = {
                 $(element.readAttribute('id')+'_'+group).show();
 
             } else {
-                template = '<div id="'+element.readAttribute('id')+'_'+group+'" class="grid tier form-list">'+template+'</div><div id="'+element.readAttribute('id')+'_'+group+'_advice"></div';
+                template = '<div id="'+element.readAttribute('id')+'_'+group+'" class="grid tier form-list">'+template+'</div><div id="'+element.readAttribute('id')+'_'+group+'_advice"></div>';
```
Comment (LeeSaferite) [inline app/design/adminhtml/default/default/template/catalog/product/edit/options/option.phtml]: This is **not** a phpdoc change. Please move to another PR.

**Comment by tmotyl** [comment]
Hi
Thanks for the review, I will split it into separate PR's to ease the review process and rebase on top of the last patch from @colinmollenhour . Closing for now

**Comment by tmotyl** [comment]
First chunk is here: https://github.com/OpenMage/magento-lts/pull/629
